File: Python3/Medium/Accepted/delete-and-earn_0.py
class Solution:
    def deleteAndEarn(self, nums: List[int]) -> int:
        
        n = max(nums)
        dp = [0]*(n+1)
        for num in nums:
            dp[num] += num
        for i in range(2, n+1):
            dp[i] = max(dp[i-1], dp[i-2]+dp[i])
        return dp[n]

File: Rust/Medium/WrongAnswer/delete-and-earn_0.rs
impl Solution {
    pub fn delete_and_earn(nums: Vec<i32>) -> i32 {
        
        let mut dp = vec![0; 10001];
        for num in nums {
            dp[num as usize] += num;
        }
        for i in 2..10001 {
            dp[i] = dp[i].max(dp[i-1]);
            dp[i] = dp[i].max(dp[i-2] + dp[i]);
        }
        dp[10000]
    }
}